jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/SensuWriterFactory.java

```diff
@@ -48,40 +48,56 @@
 	@Nonnull private final InetSocketAddress server;
 	@Nonnull private final ImmutableList<String> typeNames;
 	@Nullable private final String rootPrefix;
+	@Nonnull private final String handler;
+	@Nonnull private final String name;
+	@Nonnull private final String type;
 	@Nonnull private final FlushStrategy flushStrategy;
 	private final int poolSize;
 
 	public SensuWriterFactory(
-			@JsonProperty("typeNames") ImmutableList<String> typeNames,
-			@JsonProperty("booleanAsNumber") boolean booleanAsNumber,
-			@JsonProperty("host") String host,
-			@JsonProperty("port") Integer port,
-			@JsonProperty("rootPrefix") String rootPrefix,
-			@JsonProperty("flushStrategy") String flushStrategy,
-			@JsonProperty("flushDelayInSeconds") Integer flushDelayInSeconds,
-			@JsonProperty("poolSize") Integer poolSize) {
-		this.rootPrefix = rootPrefix;
-		this.typeNames = firstNonNull(typeNames, ImmutableList.<String>of());
+		@JsonProperty("booleanAsNumber") boolean booleanAsNumber,
```

Comment:
Why did you reorder arguments?
Except that LGTM

Reply:
Ordering is now back to normal, with my 3 additions added at the bottom, in the order they are called in (handler -> name -> type)